### PR TITLE
Add scripts to create folder and import alert-rules in grafana

### PR DESCRIPTION
**Pull request by klesaulnier**

### CODE REVIEW

File: constant.py [hunk 1]
```diff
@@ -23,8 +23,12 @@
 HTTP_PROTOCOL = "http://"
 API_VERSION = "/v1"
 
+GRAFANA_API_VERSION = "/v1"
+GRAFANA_API = "/api"
+GRAFANA_PROVISIONING = "/provisioning"
+
 # TODO add command parameter to set this DEV
-DEV = False
+DEV = True
```
Comment (antoinebhs) [inline constant.py:31]: Should be left to false

File: README.md [hunk 2]
```diff
@@ -26,3 +26,45 @@ Delete Loadflow results definitively :
...
+```py
+python3 grafana_create_folder.py {{FOLDER_NAME}}
+```
+
+This command should return a json with a field "uid" which will be {{FOLDER_UID}} for the next command
+
+Then we can create alert-rules in the previously created folder.
+Execute this command :
+```py
+python3 grafana_create_alerts.py -f {{PATH_TO_ALERT_RULES_1}} -f {{PATH_TO_ALERT_RULES_2}} -d {{DATASOURCE_UID}} -p {{FOLDER_UID}}
```
Comment (antoinebhs) [inline README.md:43]: We can document "-i" the rule group interval?

File: README.md [hunk 3]
```diff
@@ -26,3 +26,45 @@ Delete Loadflow results definitively :
...
+```
+
+*Note: **DATASOURCE_UID** can be found in URL by editing a datasource in grafana UI*
```
Comment (antoinebhs) [inline README.md:46]: We can add also that if the folder already exists, we can simply get the folder uid in the url by clicking the folder in the Dashboards menu.

File: constant.py [hunk 4]
```diff
@@ -43,6 +47,8 @@
 SHORTCIRCUIT_SERVER_HOSTNAME = "localhost:5031" if DEV else "shortcircuit-server"
 VOLTAGE_INIT_SERVER_HOSTNAME = "localhost:5038" if DEV else "voltage-init-server"
 
+GRAFANA_HOSTNAME = "localhost:7000" if DEV else "grafana"
```
Comment (antoinebhs) [inline constant.py:50]: Are we sure about "grafana"?

File: functions/grafana/create_alert.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    print("Importing alert-rule " + alert_rule_json_path)
+    file_content = Path(alert_rule_json_path).read_text()
+
+    file_content = file_content.replace(FOLDER_UID_PLACEHOLDER, parent_folder_uid)
+    file_content = file_content.replace(DATASOURCE_UID_PLACEHOLDER, datasource_uid)
+
+    file_content_json = json.loads(file_content)
+
+    try:
+        result = requests.post(constant.GRAFANA_ALERT_RULES, json = file_content_json, headers={'X-Disable-Provenance': 'true'})
```
Comment (antoinebhs) [inline functions/grafana/create_alert.py:31]: We should document maybe in the README? that we use X-Disable-Provenance': 'true' to make the alerts editable

**Review by antoinebhs — CHANGES_REQUESTED** [review]
See above comments for changes.
Good job! It works well locally, we should think about authentication and try it on our devopf cluster as the next step

File: constant.py (hunk 1 above)
Comment (klesaulnier) [reply constant.py:31]: fixed, I'll always miss this one I guess

File: README.md (hunk 2 above)
Comment (klesaulnier) [reply README.md:43]: fixed

File: README.md (hunk 3 above)
Comment (klesaulnier) [reply README.md:46]: fixed

File: functions/grafana/create_alert.py (hunk 5 above)
Comment (klesaulnier) [reply functions/grafana/create_alert.py:31]: changed

File: functions/grafana/create_folder.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import constant
+
+#
+# @author Kevin Le Saulnier <kevin.lesaulnier at rte-france.com>
+#
+
+def create_folder(folder_name):
+    folder_data = {"title": folder_name}
+    try:
+        result = requests.post(constant.GRAFANA_FOLDER, json = folder_data)
```
[the remark this answers is not in the document]
Comment (klesaulnier) [reply functions/grafana/create_folder.py:18]: I don't think this is a good idea, uids are only technical, and forcing them could cause troubles if we target already existing ones (for exemple in production)
I don't see any gain doing so

**Review by antoinebhs — APPROVED** [review]

**Pull request merged**